### DISCUSSION TITLE
Integration proposal/New feature: Object distance from mouse click

### DISCUSSION THREAD

**Discussion by aaronlsmiles** [decadenza/SimpleStereo#18] [Ideas]
Once we have the depth map it would be really useful to be able to click on an object in the scene and obtain its distance from the cameras.

[This code](https://github.com/aaronlsmiles/StereoVision/blob/master/LTwU-Stereo-Vision-master/Main_Stereo_Vision_Prog.py#L29) used to do it, but I can't get it to run since its build on deprecated versions of python/opencv. 

Need to get the following code blocks ported in to work in SimpleStereo somehow/where.

```
def coords_mouse_disp(event,x,y,flags,param):
    if event == cv2.EVENT_LBUTTONDBLCLK:
        #print x,y,disp[y,x],filteredImg[y,x]
        average=0
        for u in range (-1,2):
            for v in range (-1,2):
                average += disp[y+u,x+v]
        average=average/9
        Distance= -593.97*average**(3) + 1506.8*average**(2) - 1373.1*average + 522.06
        Distance= np.around(Distance*0.01,decimals=2)
        print('Distance: '+ str(Distance)+' m')
```

[Then](https://github.com/aaronlsmiles/StereoVision/blob/master/LTwU-Stereo-Vision-master/Main_Stereo_Vision_Prog.py#L244),
```
    # Mouse click
    cv2.setMouseCallback("Filtered Color Depth",coords_mouse_disp,filt_Color)
```

**Comment by decadenza** [comment]
This question is not related to SimpleStereo. I suggest you to search and post it over general forums like StackOverflow.

**Reply by aaronlsmiles** [reply]
Correct, it's not a question... it's a feature request. It is related specifically to SimpleStereo. The readme says to post feature requests and this Discussion board has a dropdown specifically for it!? 😕

https://github.com/decadenza/SimpleStereo#contributions

> "proposing integrations [] is highly encouraged"

![2022-10-13](https://user-images.githubusercontent.com/60931145/195490184-4b699325-00a9-4cd4-993e-251631da1856.jpg)

**Reply by aaronlsmiles** [reply]
We're working together here, or at least I thought/hoped so! I was just trying to think of features that would make this package more appealing and attract more people, is that not the goal?

**Reply by decadenza** [reply]
Sorry, did not see the category. You are free to do a pull request.

**Reply by aaronlsmiles** [reply]
Thanks! I've created a pull request to the experimental branch with comments.
